### PR TITLE
othermetas.js: Add /stone

### PR DESCRIPTION
**Pull request by KrisOnline**
I couldn't use `Chat.getDataPokemonHTML`, `Chat.getDataItemHTML`, or `Chat.getDataMoveHTML`, as this required elements that none of those fully provide.

### CODE REVIEW

**Comment by KrisOnline** [comment]
Here's what they look like:
Example

**Comment by KrisOnline** [comment]
(oh also this was requested by Chloe, who leads Mix and Mega and is an OM room RO)

File: chat-plugins/othermetas.js [hunk 1]
```diff
@@ -147,6 +147,109 @@ exports.commands = {
...
+		if (megaTemplate.types.length > baseTemplate.types.length) {
+			deltas.type = megaTemplate.types[1];
+		} else if (megaTemplate.types.length < baseTemplate.types.length) {
+			deltas.type = baseTemplate.types[0];
+		} else if (megaTemplate.types[1] !== baseTemplate.types[1]) {
+			deltas.type = megaTemplate.types[1];
+		}
+		let details = {
+			"Gen": 6,
+			"Weight": (JSON.stringify(deltas.weightkg).startsWith("-") ? "" : "+") + Math.round(deltas.weightkg * 100) / 100 + " kg",
```
Comment (urkerab) [inline chat-plugins/othermetas.js]: What was wrong with `deltas.weightkg > 0`? (or `>= 0`, but personally I'd prefer "0 kg" for stones that don't change the user's weight)
Nit: weights are currently accurate to the nearest 0.1 kg

File: chat-plugins/othermetas.js [hunk 2]
```diff
@@ -147,6 +147,109 @@ exports.commands = {
...
+			baseTemplate = Dex.getTemplate("Groudon");
+			megaTemplate = Dex.getTemplate("Groudon-Primal");
+		} else if (stone.id === 'blueorb') {
+			baseTemplate = Dex.getTemplate("Kyogre");
+			megaTemplate = Dex.getTemplate("Kyogre-Primal");
+		}
+		let deltas = {
+			baseStats: {},
+			weightkg: megaTemplate.weightkg - baseTemplate.weightkg,
+		};
```
Comment (urkerab) [inline chat-plugins/othermetas.js]: I looked into calling `getMegaDeltas` but the best I could do was `Dex.forFormat('gen7mixandmega').dataCache.Scripts.getMegaDeltas.call(Dex, megaTemplate)` which seems suboptimal.

File: chat-plugins/othermetas.js [hunk 3]
```diff
@@ -147,6 +147,109 @@ exports.commands = {
...
+			tier = "Stone";
+		}
+		let buf = `<li class="result">`;
+		buf += `<span class="col numcol">${tier}</span> `;
+		if (targetid === "dragonascent") {
+			buf += `<span class="col itemiconcol"></span>`;
+		} else {
+			buf += `<span class="col itemiconcol"><psicon item="${targetid}"/></span> `;
+		}
+		if (targetid === "dragonascent") {
```
Comment (urkerab) [inline chat-plugins/othermetas.js]: (Seems a little odd to have two `if`s for the same condition in a row...)

File: chat-plugins/othermetas.js [hunk 4]
```diff
@@ -147,6 +147,109 @@ exports.commands = {
...
+		buf += `<span style="float:left;min-height:26px">`;
+		buf += `<span class="col statcol"><em>HP</em><br />0</span> `;
+		buf += `<span class="col statcol"><em>Atk</em><br />${deltas.baseStats.atk}</span> `;
+		buf += `<span class="col statcol"><em>Def</em><br />${deltas.baseStats.def}</span> `;
+		buf += `<span class="col statcol"><em>SpA</em><br />${deltas.baseStats.spa}</span> `;
+		buf += `<span class="col statcol"><em>SpD</em><br />${deltas.baseStats.spd}</span> `;
+		buf += `<span class="col statcol"><em>Spe</em><br />${deltas.baseStats.spe}</span> `;
+		buf += `<span class="col bstcol"><em>BST<br />100</em></span> `;
+		buf += `</span>`;
+		buf += `</li>`;
```
Comment (urkerab) [inline chat-plugins/othermetas.js]: Yes, this is annoying. At one point I was toying with the idea of having a single function in Chat which would look for well-known properties of the effect passed in and generate appropriate details. I wonder whether I should dig it out and update it to work with a set of mega deltas...

File: chat-plugins/othermetas.js [hunk 5]
```diff
@@ -147,6 +147,109 @@ exports.commands = {
 	},
 	mixandmegahelp: ["/mnm <pokemon> @ <mega stone> - Shows the Mix and Mega evolved Pokemon's type and stats."],
 
+	'!stone': true,
+	move: 'stone',
```
Comment (Spandamn) [inline chat-plugins/othermetas.js]: This should not be an alias, /move already exists

Comment (Zarel) [reply chat-plugins/othermetas.js]: Oh yeah it does what are you doing overriding core sim features???

Comment (KrisOnline) [reply chat-plugins/othermetas.js]: oOPS

**Pull request merged**